### ISSUE TITLE
Re-enable parallel jobs on VagrantPlaybookCheck / infra-vagrant-1

### ISSUE THREAD

**Issue by sxa** [AdoptOpenJDK/openjdk-infrastructure#1287]
Now that Windows VMs in Vagrant are looking more stable after [this change](https://github.com/AdoptOpenJDK/openjdk-build/pull/1689), we should see if we can get the parallelism up since it takes a long time at the moment to test across all platforms. We can use this issue to log progress and what issues we have trying to turn it back on.

We had this [enabled previously](https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/923) but had to turn it off again as things were very unstable with it enabled.

**Comment by Willsparker** [comment]
Would you mind increasing the executors on `infra-vagrant-1` ? I suppose we could see if having `2` shows the issues we had before? :-)

**Comment by sxa** [comment]
I have granted @Willsparker configure access to [infra-vagrant-1](https://ci.adoptopenjdk.net/computer/infra-vagrant-1/) in jenkins (he's the only person who uses it and has full access to the underlying machine already) so is able to make this change now.

**Comment by Willsparker** [comment]
Executors have been put up to 3, with a JDK8 job (fastmode, run build, but no Windows, no test)
https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/523/
Just as a test to ensure the error is still there, mainly.

**Comment by Willsparker** [comment]
A rebuild of the above run showed the issue is still there: 
https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/526/
Each of the failing OSs have failed with error messages that would point to the machine being turned off half way through execution ( i.e., `Failed to connect to the host via ssh: Shared connection to ___ closed `  or  `A prompt was encountered while running a command, but no input data was specified`)

Running `vagrant global-status --prune` shows the machines that failed, were left on the machine, so other runs of VPC aren't accidentally destroying them. Their state varies between `poweroff` and `aborted`.

It seems that the end of `ansible` commands are correlated to the failure of a different OS, i.e.
CentOS6 failed at around 22:58:00
CentOS8 finished it's build at around 22:56:36 (the build started with `ansible all .... `

Rerunning without the `build` option as the jenkins job will hang for awhile if half way through a build and the VM 'turns off'. It should give us a better idea of if this correlation is correct. If so, it may be helped by #1303

**Comment by Willsparker** [comment]
Rerunning without using `build` or `test` has given a better show of the correlation: 

| OS         | Passed / failed | Time of PB end | Time of vagrant destroy |
|------------|-----------------|----------------|-------------------------|
| Debian8    | Failed          | 09:33:33       | N/A                     |
| SUSE12     | Passed          | 09:24:53       | 9:25:04                 |
| FreeBSD12  | Passed          | **09:29:06**   | 9:29:17                 |
| CentOS6    | Failed          | **09:29:19**   | N/A                     |
| CentOS7    | Failed          | **09:29:19**   | N/A                     |
| CentOS8    | Passed          | 09:21:18       | 09:21:30                |
| Ubuntu1604 | Passed          | 10:10:08       | 10:10:19                |
| Ubuntu1804 | Failed          | ~10:03:11      | N/A                     |

It seems when `FreeBSD12` passed, it caused `CentOS6` and `CentOS7` to fail. (on `infra-vagrant-1`, `vagrant global-status --prune` shows them being in the `poweroff` state).
`Ubuntu1804` was hung in the `Running ./configure and make` on the `cmake` task (after about half an hour, compared to it taking ~9 minutes to build on `Ubuntu1604`) - this may be caused by `Ubuntu1604`s pass turning off the VM and causing the job to hang (`Ubuntu1804` was also in the `poweroff` state).
Looking at the times, it's not immediately obvious what caused `Debian8` to fail. Interestingly, `Debian8` is in the `aborted` state on `infra-vagrant-1`.

**Comment by Willsparker** [comment]
@sxa and I have determined that the cause of the issue is the `vagrant halt` command - it seems to be halting all the VMs on the machine, not just the intended one. Working on trying to get a log that has the output of `vagrant halt --debug`, specifically when the issue occurs, as this is intermittent.
The particular Virtualbox command that's being ran to turn off the vagrant machines is:
` VBoxManage controlvm <VM_UUID> poweroff `. If we can get a log of the output when it closes down the VMs it's not meant to, we can determine if this is a bug with vagrant, or virtualbox, itself.
If vagrant is the issue, it's possible we could just run that VBoxManage command manually as a workaround.

**Comment by Willsparker** [comment]
[VHalt_log_FreeBSD12.log](https://github.com/AdoptOpenJDK/openjdk-infrastructure/files/4563532/VHalt_log_FreeBSD12.log)

Managed to get a long from a run where the halting VM appeared to stop another VM.
https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/548/

FreeBSD seemed to stop CentOS7. Unfortunately, the log doesn't appear to be out of the ordinary from a standard log. (i.e. no additional `VBoxManage controlvm <VM_UUID> poweroff`)

Next thing to test is turning the VMs off using that command manually.

**Comment by Willsparker** [comment]
https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/550/

Unfortunately, using the `VBoxManage` commands would cause the same issue - It appears CentOS7 'turned off' at the time SUSE12 ran the command. This would point to it being a Virtualbox issue. It appears the VMs that are erroring are going into an `aborted` state. 
At current, according to Jenkins, `SUSE12` and `CentOS8` have succeeded, `CentOS7` has failed and `FreeBSD12`, `CentOS6` and `Ubuntu1604` are running.
However, this is the current output of `infra-vagrant-1` running `vagrant global-status --prune`
```
69e9457  adoptopenjdkC8     virtualbox poweroff
4077d0c  adoptopenjdkS12    virtualbox poweroff
d5250a9  adoptopenjdkC7     virtualbox aborted
4fde107  adoptopenjdkFBSD12 virtualbox aborted
6c4065b  adoptopenjdkC6     virtualbox running
b415a74  adoptopenjdkU16    virtualbox running
```
This broadly makes sense as FreeBSD appears to have hanged, looking at the Jenkins console output - thereby implying it too has failed like the CentOS7 VM.

Some Googling shows an issue describing similar circumstances : https://www.virtualbox.org/ticket/8582
However this is from nearly a decade ago, with no solution. Beyond describing a similar problem, not much can be gathered from it so.. moving on.

The current version of Virtualbox / VBoxManage on `infra-vagrant-1` is `6.0.14r133895`. As I've only seen this issue on `infra-vagrant-1`, I think the next logical step is to try re-installing Virtualbox. Virtualbox is currently at version `6.1.6`, so I may as well try that.

**Comment by Willsparker** [comment]
Vagrant `2.2.6` supports Virtualbox `6.0` and less - It wasn't until `2.2.7` that it it officially supported Virtualbox `6.1`. Therefore, I've chosen to upgrade to `6.0.20-137117` instead, to change as few variables as possible

**Comment by Willsparker** [comment]
Okay, from the looks, upgrading (and reinstalling) Virtualbox wasn't the answer, looking at https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/551/ . I ended up just upgrading to `6.0.20r137117` as the current version of vagrant (`2.2.6`) is incompatible with VirtualBox `6.1`, I thought it'd be better to not change too much in one go.

**Comment by Willsparker** [comment]
A look at the Virtualbox logs also don't provide any insight. Here are the logs for https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/553/

[FBSD12.VBox.S.log](https://github.com/AdoptOpenJDK/openjdk-infrastructure/files/4574161/FBSD12.VBox.S.log)
[U16.VBox.F.log](https://github.com/AdoptOpenJDK/openjdk-infrastructure/files/4574164/U16.VBox.F.log)

The FreeBSD Vm succeeded from the run. When looking at the log (or `grep`ing "VM State"), there's a progression of states that it goes through: 
```
Changing the VM state from 'CREATING' to 'CREATED'
Changing the VM state from 'CREATED' to 'POWERING_ON'
Changing the VM state from 'POWERING_ON' to 'RUNNING'
Changing the VM state from 'RUNNING' to 'POWERING_OFF'
Changing the VM state from 'POWERING_OFF' to 'OFF'
Changing the VM state from 'OFF' to 'DESTROYING'
Changing the VM state from 'DESTROYING' to 'TERMINATED'
```

However, Ubuntu1604 log contains only VM state changes up to `Changing the VM state from 'POWERING_ON' to 'RUNNING'` , with the log seemingly being cut off at the end.

**Comment by Willsparker** [comment]
I attempted to recreate the issue by using `vagrant halt` and the `VBoxManage` equivalent, on the `infra-vagrant-1` machine, as `VPC` ran. Unfortunately, I was unable to recreate the issue by running these commands manually, despite trying 3 times each. This would point to it being a Jenkins specific issue- I'm going to try and recreate this issue on the `infra-vagrant-T541` with 3 executors, as that would rule it out being a machine specific issue

**Comment by Willsparker** [comment]
Having ran `VPC` on `infra-vagrant-T541` last night ( https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/557/ ), the issue still cropped up, so it's not a machine specific issue.

| OS         | Pass / Fail | Time       |
|------------|-------------|------------|
| FreeBSD12  | Fail        | 00:10:06   |
| Ubuntu1604 | Pass        | 23:47:36   |
| Ubuntu1804 | Pass        | 23:44:05   |
| CentOS6    | Pass        | 00:12:57   |
| CentOS7    | Fail        | 00:12:54   |
| CentOS8    | Fail        | _00:34:24_|
| SUSE12     | Fail        | _00:58:22_ |

 ** The OSs with times in italic failed due to a playbook issue, not the parallelisation issue.

It's likely that CentOS6 passing is what caused CentOS7 to shutdown. It possibly also caused FreeBSD12 to shutdown (however, I can only speculate as the time above reflects when the last build package was installed. The error message was several hours later due to ansible hanging).

I'm going to try a run of `VPC` without shutting down the VMs at all to determine if that would be a viable workaround.

**Comment by Willsparker** [comment]
Over the last few days, several `VPC` runs have been done in an attempt to find a situation where the problem isn't there. I've stripped out all of the vagrant commands, so now the only thing affecting the boxes should be `vagrant up`, however the problem still persists. As well as that, the log files from `vagrant up --debug` don't provide much insight into what's happening to the other VMs.
Example `vagrant up --debug` log file from [run 565:](https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/565/) 

[vup.Ubuntu1604.2005061014.log](https://github.com/AdoptOpenJDK/openjdk-infrastructure/files/4591476/vup.Ubuntu1604.2005061014.log)

_If_ `vagrant up` is the command that's causing the VMs to close, then Ubuntu1604 will be the most likely cause of Ubuntu1804 and CentOS6 closing - 
 U18 fail time: `16:13:38` - `16:50:58`
 C7 fail time: `16:13:40` - `16:25:07`
These times both include hanging, but are most likely closer to the first number as the tasks they were running should only take a couple minutes each.
 U16 vagrant up time: `16:14:51`

However, another time that had relevance was SUSE12 finishing it's run at `16:13:37` - i.e. both the ansible playbook finishes AND the Jenkins run finishes. As I haven't been able to recreate the issue not through a Jenkins connection, AND ansible uses ssh to run the commands, both of them being issues could be possible. As counter-intuitive as it is, I'm going to try a run of `VPC` where the playbook doesn't run, but instead runs `sleep 100`, to see if the problem persists.

**Comment by Willsparker** [comment]
Looks like I've found the issue.
Turns out, it was the Jenkins link. When Jenkins finishes running a job, it stops all processes spawned by the job ( https://wiki.jenkins.io/display/JENKINS/ProcessTreeKiller ), and determines that by looking for environment variables that persist in a child process when spawned from a parent process. As far as I can speculate, the same Jenkins job were starting these Vagrant VMs, so the processes running the VMs would contain the same variables and all (or some) the VMs running would be destroyed when one OS finishes it's run - thereby ruining other OSs. 
The way to stop this is to start the VMs running `BUILD_ID=dontKillMe vagrant up`. 

Runs where this worked:
https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/567/
https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/568/

There may be some additional commands that will need this setting, but that can be resolved with later testing.

**Comment by Willsparker** [comment]
Unfortunately have to reopen this as parallel jobs have decided to break again.
https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/639/
^ This is by far not the first time it failed again, but it's the first time I've been able to find a similar situation, in which it looks like on OS failing/Passing, causes others to fail:

```
Win2012 : P : 	19:37:08		<<<
Deb8 	: F : 	15:35:32 - 15:36:34	<<
Deb10 	: F :	15:35:08 - 15:36:39	<<
FBSD12 	: P :	17:06:55		<
U16 	: F : 	17:06:51		<
U18 	: F : 	15:30:35 - 15:36:34	<<
C6 	: F : 	17:05:44 - 19:17:57	<
C7 	: F : 	17:06:28 - 19:17:57	<
C8 	: F : 	19:35:12 - 21:46:46	<<<
SUSE12 	: F : 	15:35:34 - 15:35:39	<< (Failed due to GIT_SOURCE error)
```
i.e.: 
FBSD caused all the VMs w/ `<` to shutdown
SUSE12 caused all VMs w/ `<<` to shutdown
Win2012 caused all VMs w/ `<<<` to shutdown

**Comment by Willsparker** [comment]
That particular `VPC` was testing this branch, at this commit : https://github.com/Willsparker/openjdk-infrastructure/commit/bc1a5a34cdb6f29eb82daa8a2d5187c5b8f620c9
therefore, all `vagrant up`s had `BUILD_ID=dontKillMe` on them.
In addition, the Jenkins job is configured to start running `./vagrantPlaybookCheck.sh ...` with `BUILD_ID=dontKillMe` at the beginning.

**Comment by Willsparker** [comment]
Through `VPC`, I created several separate runs (runs 659 to 668), testing a single OS, to determine the current state of the playbooks. From this I found:
- CentOS6 ([662](https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/662/console)) and CentOS7([661](https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/661/console)) had both hung in the early stages of the playbook. The machine the VMs were running on showed them still running, however taking very little CPU. Once I stopped their respective runs, the VMs were shutdown instantly. This would suggest a similar issue to the one before, where Jenkins kills the process once the run is over.

- Despite running one at a time, the Ubuntu1604 ([664](https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/OS=Ubuntu1804,label=vagrant/664/)) and Ubuntu1804 ([663](https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/OS=Ubuntu1804,label=vagrant/663/))VMs also failed, showing the `Failed to connect to the host via ssh` issue that shows when the VMs are shutdown. Unfortunately I wasn't around to check the host machine when this was happened, however, the next morning, both of the VMs are still there, however U18 shows a `poweroff` state and U16 shows an `aborted` state, according to `vagrant global-status --prune`. The error message would suggest that the VMs were shutdown before the end of the Jenkins run, so something (possibly on the host machine) caused the ssh connection to break. 

- Windows 2012r2, CentOS8, Debian 8 and Debian 10 succeeded when running one at a time. Therefore whatever affected the Ubuntu boxes above could be intermittent.

**Comment by Willsparker** [comment]
Currently running on node `infra-vagrant-T541` running Linux Mint w/ 2 build executors.
https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/672/
https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/673/

It appears running the job on this machine is significantly more stable, with the issues being that the Jenkins agent disconnected (probably due to finnicky home internet connection, as the laptop is at my house) and the `Ant-Contrib` download being unstable, again, possibly due to my home internet connection. 

Thankfully this would rule out Jenkins being the issue and would instead point to an issue with the node that runs this job, `infra-softlayer-ubuntu1804-x64-1` (AKA `infra-vagrant-1`).

**Comment by Willsparker** [comment]
The `vagrantPlaybookCheck.sh` script was run locally on the `infra-vagrant-1` machine for CentOS6/7 and Debian 8. While both CentOS runs succeeded, Debian failed with `Failed to connect to the host via ssh`. This means the instability of the VPC is confirmed to be an issue with the machine itself, not Jenkins.

Up until now, 2 QEMU RISC-V VMs have been hosted on the machine, one of which was connected to Jenkins ([test-sxa-fedora28-riscv-1](https://ci.adoptopenjdk.net/computer/test-sxa-fedora28-riscv-1/)). I have temporarily stopped those VMs so I can get a run of `VPC` without those interfering. The current working theory is that the machine is unable to cope with using 2 pieces of software (vagrant and QEMU) that both use Virtualbox underneath to run them (The machine showed several duplicates of the `VBoxSVC` process running, where I believe there's only meant to be one). 

VPC Run : https://ci.adoptopenjdk.net/job/VagrantPlaybookCheck/676/

**Comment by Willsparker** [comment]
From the looks of the above run, it's better than it was at the end of last week, however I still think the problem is there as several VMs weren't able to make an SSH connection, and the CentOS7 run had to be stopped as it had hung. However all of these VMs were still alive and I was able to ssh into them from the host machine after the run had ended, therefore the only issue is now that the ssh connection is being broken. Next VPC Run for confirmation : https://ci.adoptopenjdk.net/job/VagrantPlaybookCheck/677/

**Comment by Willsparker** [comment]
Due to the above run coming up with a lot of `ssh connection closed` errors, I'm convinced it's something wrong `VirtualBox` on the machine. The plan is to uninstall the current version `v6.0.20` and put the version that's on the working machine onto it- `v6.0.14`, Neither of which are particularly old.
Unfortunately uninstall Virtualbox is proving to be fairly tricky:
```
 root@infra-softlayer-vagarant-1:~# dpkg -r virtualbox-6.0
(Reading database ... 176841 files and directories currently installed.)
Removing virtualbox-6.0 (6.0.20-137117~Ubuntu~bionic) ...
dpkg: error processing package virtualbox-6.0 (--remove):
 installed virtualbox-6.0 package pre-removal script subprocess returned error exit status 1
/var/lib/dpkg/info/virtualbox-6.0.postinst: 106: /var/lib/dpkg/info/virtualbox-6.0.postinst: /usr/lib/virtualbox/prerm-common.sh: not found
/var/lib/dpkg/info/virtualbox-6.0.postinst: 107: /var/lib/dpkg/info/virtualbox-6.0.postinst: /usr/lib/virtualbox/postinst-common.sh: not found
Errors were encountered while processing:
 virtualbox-6.0
```
During this, a message comes up informing me:
```
 Running VMs found                                                                                                                                                                    
 VirtualBox is currently running. Please close it and try again. Please note that it can take up to ten seconds for VirtualBox (in particular the VBoxSVC daemon) to finish running.
```
However, I distinctly see no virtualbox processes running:
```
root@infra-softlayer-vagarant-1:~# ps aux | grep VBox
root      2863  0.0  0.0  15500  2216 pts/0    S+   05:58   0:00 grep --color=auto VBox
root@infra-softlayer-vagarant-1:~# ps aux | grep vbox
root      2869  0.0  0.0  15500  2088 pts/0    S+   05:58   0:00 grep --color=auto vbox
root@infra-softlayer-vagarant-1:~# ps aux | grep virtualbox
root      2891  0.0  0.0  15500  2216 pts/0    S+   05:59   0:00 grep --color=auto virtualbox
```
Which is mildly infuriating

**Comment by Willsparker** [comment]
Solution found! https://askubuntu.com/a/835407

It seems because I attempted to `manually` uninstall `VBox` by deleting `/usr/lib/virtualbox`, it couldn't find those scripts in the error message above and subsequently failed. However, those scripts don't _really_ do anything it seems, so a way to fix it is to just recreate them, make them executable and run 
`apt rm virtualbox-6.0` (or `dpkg -r virtualbox-6.0`) :
```
root@infra-softlayer-vagarant-1:# apt remove virtualbox-6.0
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages will be REMOVED:
  virtualbox-6.0
0 upgraded, 0 newly installed, 1 to remove and 151 not upgraded.
After this operation, 228 MB disk space will be freed.
Do you want to continue? [Y/n] Y
(Reading database ... 176841 files and directories currently installed.)
Removing virtualbox-6.0 (6.0.20-137117~Ubuntu~bionic) ...
Processing triggers for shared-mime-info (1.9-2) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for mime-support (3.60ubuntu1) ...
```
It should also be mentioned that in this uninstallation, if a complete purge is required (i.e. No user settings left, no virtualboxes left over), for each user, run:
```
rm -rf ~/.config/virtualbox
rm -rf ~/Virtualbox\ VMs
```
The version of Virtualbox to be reinstalled was from [here](https://www.virtualbox.org/wiki/Download_Old_Builds_6_0). I specifically used the Ubuntu 18.04 / 18.10 / 19.04 .deb file of VBox `6.0.14` and installed it via:
```
root@infra-softlayer-vagarant-1:~# dpkg -i virtualbox-6.0_6.0.14-133895~Ubuntu~bionic_amd64.deb 
Selecting previously unselected package virtualbox-6.0.
(Reading database ... 176015 files and directories currently installed.)
Preparing to unpack virtualbox-6.0_6.0.14-133895~Ubuntu~bionic_amd64.deb ...
Unpacking virtualbox-6.0 (6.0.14-133895~Ubuntu~bionic) ...
Setting up virtualbox-6.0 (6.0.14-133895~Ubuntu~bionic) ...
addgroup: The group `vboxusers' already exists as a system group. Exiting.
Processing triggers for systemd (237-3ubuntu10.33) ...
Processing triggers for ureadahead (0.100.0-21) ...
ureadahead will be reprofiled on next reboot
Processing triggers for mime-support (3.60ubuntu1) ...
Processing triggers for hicolor-icon-theme (0.17-2) ...
Processing triggers for shared-mime-info (1.9-2) ...

root@infra-softlayer-vagarant-1:~# VBoxManage --version
6.0.14r133895
```

**Comment by Willsparker** [comment]
Unfortunately the reinstallation of VBox didn't help the issue in the `VPC` run. From the extra debug shown, U18 and FBSD12 were turned off between `13:10:10` and `13:43:28`. I've added another `vagrant global-status --prune` to the beginning of the Jenkins Job to see if it's between Jenkins Jobs ending and starting that's causing the issue.

**Comment by Willsparker** [comment]
@sxa and I investigated yesterday. It appears that there are 2 issues that are preventing clean `VPC` runs- One where there are multiple `VBoxNetDHCP` processes, the VMs get confused and can change IP addresses halfway through `VPC` running, causing the machine to stay powered on, but the playbook fails. Manually `kill`ing all the extra VBoxNetDHCP seems to stop this, however we're keeping an eye on how these extra processes happen. 
If all else fails, we could investigate setting a static IP to each VM ( https://www.vagrantup.com/docs/networking/private_network.html#static-ip ) however this will come with having to be very diligent in making sure old VMs are cleared up before starting new ones.

The other issue appears to be similar to the ProcessTreeKiller issue above - VMs will seemingly close down when other OSs finish their runs- we're investigating setting `BUILD_ID` to different things for each run, and adding it to other commands (such as the ssh commands), to see if that changes anything

**Comment by Willsparker** [comment]
I was able to setup a separate Jenkins instance and run `VPC` on it without the issues occurring on it - this was with Jenkins `2.235.1` and with the cmd line argument to specifically disable `ProcessTreeKiller` : 
```
java -Dhudson.util.ProcessTree.disable=true -jar jenkins.war
```
I've got a list of situations I can try to confirm the circumstances in which these issues arise, which could possibly lead to a fix. (The first in the list is the aforementioned run)

- [X] Jenkins `2.235.1`, `ProcessTreeKiller` disabled
Outcome: No VMs shutdown
- [ ] Jenkins `2.235.1`, don't disable `ProcessTreeKiller`. This will be the standard once #1414 is fixed
Outcome: 
- [ ] Jenkins `2.222.4`, don't disable `ProcessTreeKiller`
Outcome:
- [ ] Jenkins `2.222.4`, `ProcessTreeKiller` disabled
Outcome:

**Comment by Willsparker** [comment]
All of the above runs ran through cleanly. This would suggest that it's not the Jenkins version updating that is causing the issues, and that the `BUILD_ID` changes @sxa and I have made work to stop `ProcessTreeKiller`.
It also confirms that the machine running `VPC` isn't at fault.

I'm rerunning the above test, but instead of having `VPC` run on the master node, I've connected a slave node (what is called `infra-vagrant-t541` on ci.adoptopenjdk.net) to my standalone Jenkins instance to see if that adds any instability.

- [ ] Jenkins 2.222.4, ProcessTreeKiller disabled
- [ ] Jenkins 2.222.4, don't disable ProcessTreeKiller
- [ ] Jenkins 2.235.1, don't disable ProcessTreeKiller
- [ ] Jenkins 2.235.1, disabled `ProcessTreeKiller`

**Comment by sxa** [comment]
Jenkins is killing the `VBoxXPCOMIPCD` process which manages inter-process communication between the VirtualBox process which is causing it to shutdown the VM.

**Comment by karianna** [comment]
I need to try that Jenkins restart :-/

**Comment by sxa** [comment]
Unfortunately I don't think that will resolve this one :-(

**Comment by Willsparker** [comment]
@sxa How'd you find it was the `VBoxXPCOMIPCD` process?

**Comment by sxa** [comment]
Have also raised https://issues.jenkins-ci.org/browse/JENKINS-62923 against jenkins

**Comment by sxa** [comment]
> @sxa How'd you find it was the `VBoxXPCOMIPCD` process being killed?

By using the systemtap script that we discussed before you went on holiday to trace the kill system calls

**Comment by Willsparker** [comment]
Oh yeah! Entirely forgot we used that. Thanks for raising the issue with the Jenkins people.

**Comment by Willsparker** [comment]
@sxa came up with an idea to setup the `infra-softlayer-ubuntu1804-x64-1` machine with multiple different Jenkins agents, all with a single executor (these are at `ci.adoptopenjdk.net/computer` as `infra-softlayer-ubuntu1804-x64-1.X` )  - therefore Jenkins killing `VBoxXPCOMIPCD` isn't an issue as each VM has it's own unique process. 

This was initially causing `vagrant` to assign the same IP addresses to different machines as one User's vagrant VMs was unable to see another User's vagrant VMs, to avoid these clashes. To remedy this, I altered the `vagrantPlaybookCheck.sh` script so it instead uses `localhost:port` for ansible to connect to the machine (details of the changes can be seen at https://github.com/Willsparker/openjdk-infrastructure/tree/vagrant_port_ansible ). This fixes the issue as ports aren't user specific so one user's vagrant process doesn't need to see the other user's vagrant process. Occasionally when 2 VMs were booted at the same time, a race condition would happen where they'd be assigned the same port, however this can be avoided by ensuring there's enough `sleep` between each OS.

While this workaround to the 'Jenkins killing `VBoxXPCOMIPCD`' issue is memory intensive as running multiple jenkins agents takes up resources, I've been able to run 3 VPC OSs in parallel without issue, so this will probably suffice in the meantime whilst the above Jenkins issue is being resolved.
